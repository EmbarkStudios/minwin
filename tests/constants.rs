mod utils;
use utils::*;

test!(
    regular,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.Graphics.OpenGL.GLU_BEGIN",
        "Windows.Win32.Graphics.OpenGL.GLU_CCW",
        "Windows.Win32.Graphics.OpenGL.GL_ADD",
        "Windows.Win32.Graphics.OpenGL.GL_ALL_ATTRIB_BITS",
    ]
);

// Ensures we can generate string constants
test!(
    strings,
    [BindConfig::Minwin(Default::default()), BindConfig::Bindgen,],
    [
        // utf-16
        "Windows.Win32.Devices.Enumeration.Pnp.ADDRESS_FAMILY_VALUE_NAME",
        // ansi
        "Windows.Win32.Media.Multimedia.JOY_CONFIGCHANGED_MSGSTRING",
    ]
);

// Ensures we can generate constants that need casting to the appropriate type
// currently broken for bindgen <https://github.com/microsoft/windows-rs/issues/2504>
test!(
    casting,
    [BindConfig::Minwin(Default::default()), BindConfig::Bindgen,],
    [
        "Windows.Win32.UI.WindowsAndMessaging.IDC_SIZENESW",
        "Windows.Win32.UI.WindowsAndMessaging.IDC_SIZENS",
        "Windows.Win32.UI.WindowsAndMessaging.IDC_SIZENWSE",
        "Windows.Win32.UI.WindowsAndMessaging.IDC_SIZEWE",
        "Windows.Win32.UI.WindowsAndMessaging.IDC_UPARROW",
    ]
);
