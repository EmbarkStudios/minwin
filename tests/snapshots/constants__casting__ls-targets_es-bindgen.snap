---
source: tests/constants.rs
expression: bo.bindings
---
#![allow(non_snake_case, non_upper_case_globals, non_camel_case_types)]
pub const IDC_UPARROW: PCWSTR = 32516 as _;
pub const IDC_SIZENWSE: PCWSTR = 32642 as _;
pub const IDC_SIZENESW: PCWSTR = 32643 as _;
pub const IDC_SIZEWE: PCWSTR = 32644 as _;
pub const IDC_SIZENS: PCWSTR = 32645 as _;
pub type PCWSTR = *const u16;

